custom_components/sat/boiler.py

```
@@ -2,141 +2,382 @@
+    @property
+    def flow_return_delta(self) -> Optional[float]:
+        return self.flow_temperature - self.return_temperature if self.flow_temperature is not None and self.return_temperature is not None else None
 
-        self._last_setpoint = None
-        self._last_boiler_temperature = None
 
-    def update(self, boiler_temperature: float, boiler_temperature_derivative: float, flame_active: bool, setpoint: float):
-        """Update the tracker based on the current boiler temperature, flame status, and setpoint."""
-        if setpoint == MINIMUM_SETPOINT:
-            return
+class Boiler:
```

Pylint (4.0.4)

[refactor] Too many instance attributes (12/7)

(R0902)